src/app/app.component.ts

```diff
@@ -27,6 +32,28 @@ export class AppComponent {
     }
   }
 
+  //Atualiza a variável que indica a resolução
+  ChangeResolution(event) {
+    var newResolution = Number(event.target.value);
+    console.log(newResolution);
+    if(newResolution > 500) {
+      alert("insira uma resolução menor do que 500!");
+      event.target.value = this.Resolution;
+      return;
+    }
+    else if(newResolution < 0){
+      alert("insira uma resolução maior do que 0!");
+      event.target.value = this.Resolution;
+      return;
+    }
```

Comment:
Por que tu usou "event.target.value = this.Resolution;" nesses dois if's acima?

Reply:
Caso o usuário coloque um valor inválido de resolução ele colocar o alerta e altera o campo da resolução para a última resolução válida que o usuário digitou.

Reply:
mas "event.target.value" é essencialmente o input do usuário enviado do frontend e ele também é unidirecional, significando que alterar esse campo não vai alterar nada pro usuário. Isso serve basicamente pra leitura, não é uma variável a ser trabalhada. Inclusive se ele entrar nesses if's a resolução não é setada, não é não? Ou seja ela continua 500. Tu testou sem essas linhas e deu algum erro?